From 2e693eb80e705ea68e23eed19616d22b4778b45a Mon Sep 17 00:00:00 2001
From: James Zern <jzern@google.com>
Date: Thu, 3 Dec 2015 16:18:48 -0800
Subject: vp8: fix loop filter level clamping

the loop filter level is transmitted as 6-bits + sign so needs to be clamped in
the delta + absolute case.

BUG=https://bugzilla.mozilla.org/show_bug.cgi?id=1224363

Change-Id: Icbdca4fdbf043466429bd5c9d59dbe913bf153bc
---
 vp8/common/vp8_loopfilter.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git vp8/common/vp8_loopfilter.c vp8/common/vp8_loopfilter.c
index 8b55dff..756ad48 100644
--- vp8/common/vp8_loopfilter.c
+++ vp8/common/vp8_loopfilter.c
@@ -141,8 +141,8 @@ void vp8_loop_filter_frame_init(VP8_COMMON *cm,
             else  /* Delta Value */
             {
                 lvl_seg += mbd->segment_feature_data[MB_LVL_ALT_LF][seg];
-                lvl_seg = (lvl_seg > 0) ? ((lvl_seg > 63) ? 63: lvl_seg) : 0;
             }
+            lvl_seg = (lvl_seg > 0) ? ((lvl_seg > 63) ? 63: lvl_seg) : 0;
         }
 
         if (!mbd->mode_ref_lf_delta_enabled)
